content/dev/blog/analyzing-so-data.Rmd

```diff
@@ -0,0 +1,308 @@
+```
+
+There are a couple of pieces of this survey which illustrate some of the pain
+points faced by survey analysts. First, let's take a look at Q149 which asks
+respondents to rate how important various factors would be if they were the
+hiring manager at a major technology company. When the question is asked all of
+the possibilities are presented together:
+
+>   Q491.Congratulations! You’ve just been put in charge of technical recruiting
+>   at Globex, a multinational high-tech firm. This job comes with a corner
+>   office, and you have an experienced staff of recruiters at your disposal.
+>   All they need to be successful is a sense of the relative importance they
+>   should place on each of the following when recruiting software developers.
+>   What do you tell them?
+>
+>   * Knowledge of algorithms and data structures
+>   * Experience with specific tools (libraries, frameworks, etc.) used by the employer
+>   * Communication skills
+>   * Contributions to open source projects
+>   * Experience with specific project management tools & techniques
+>   * Previous companies worked at
+>   * Previous job titles held
+>   * Educational credentials (e.g. schools attended, specific field of study, grades earned)
+>   * Stack Overflow reputation
+
+But the limits of tabular data structures means that the answers to these
+questions are broken up into a bunch of separate variables:
+```{r Illustrate problem columns, results='asis'}
+
+vars <- schema$Column[str_detect(schema$Question,
+    "Congratulations! You've just been put in charge of technical recruiting at Globex")]
+
+stack_df[1:10, ] %>%
+    select(one_of(vars)) %>%
+    knitr::kable()
+```
+
+A better solution would be to store the responses to this question in some kind
+of tabular structure which showed the relationship between them. For instance we
+could calculate a frequency table which showed the percentage of respondents who
+gave a variable a particular rating. We can do this locally in R with something
+like the following:
+
+```{r cat array mocking, cache = TRUE, results='asis'}
+
+catarray <- stack_df[, vars] %>%
+    gather(variable, rating) %>%
+    group_by(variable, rating) %>%
+    tally() %>%
+    filter(!is.na(rating)) %>%
+    mutate(n = scales::percent(n/sum(n))) %>%
+    spread(rating, n)
+catarray$variable <- str_replace(catarray$variable, "ImportantHiring", "")
+catarray[ ,c("variable", "Very important",
+    "Important", "Somewhat important", "Not very important", "Not at all important")] %>%
+    knitr::kable()
+```
+
+This picture is great for summarizing a set of related categorical variables,
+but it's a bit of a pain to produce, and is difficult to display alongside other
+variable summaries. Since this is such a common task, Crunch has a special data
+type called a "categorical array" for working with related categorical
+variables.
+
+```{r, cache = TRUE}
+ds$importanceHiringArray <- makeArray(ds[, vars], name = "Importance to Hiring")
```

Comment:
Wow! One line does all that that you did above! Incredible!

Reply:
And maybe do a `crtabs` after this that shows essentially the same table you composed above.